src/__tests__/import-with-private-class-test.js

```diff
@@ -0,0 +1,35 @@
+const babel = require('babel-core');
+const content = `
+import * as React from "react";
+
+import type {RowProps} from "./other-row.jsx";
+
+type StaticProps = RowProps & {
+    domain: string,
+};
+
+class TableOfContentsRow extends React.Component<StaticProps> {
+    render() {
+        return <div>I'm a row</div>;
+    }
+}
+
+export default class TableOfContentsRowWithQuery extends React.Component<StaticProps> {
+    render() {
+        return <TableOfContentsRow/>;
+    }
+}
+`;
+
+it('import-object', () => {
```

Comment:
I'm not sure how necessary this test case is given there were other tests cases that were exhibiting the issue.  Let me know if you want me to remove this.

Reply:
The test is good. Let's keep it.